api-call.py

```diff
@@ -1,29 +1,226 @@
-import requests
+import datetime
 import json
 
+import pytz
+import requests
+import yaml
+
+
+def load_from_config():
+    """Loading important parameters from configuration file
+
+    :returns: The config object
+    """
+
+    with open('configuration.yaml', 'r') as file:
+        return yaml.safe_load(file)
+
+
+def log_error(error_message):
+    """Logging any error messages in the file and appending
+    with date time information in PST
+
+    :param error_message: The error message to log in file comes as a string
+    """
+
+    # Get the current datetime in PST
+    pst_timezone = pytz.timezone('America/Los_Angeles')
+    current_time = datetime.datetime.now(pst_timezone)
+    formatted_message = f"[{current_time}] {error_message}"
+
+    # Append the error message to the logfile
+    with open('logfile.txt', 'a') as file:
+        file.write(formatted_message + '\n')
+
+
+def get_access_token():
+    """Get access token from OAuth2 for connecting with OSU public APIs
+
+    :returns: The access token string
+    """
+
+    # Load configurations from the YAML file
+    config = load_from_config()
+    url = config['accessToken']['url']
+    enhancer = config['accessToken']['enhancer']
+    payload = config['accessToken']['payload']
```

Comment:
What are `enhancer` and `payload`? If `enhancer` is `Bearer`, is it any option allowed to be configured? If not, should we allow users to put other values in their config files?

Reply:
Again, as Bearer is a static string, it can be changed in future for any reason, so I put it as an enhancer of the token in the config/static parts of the application. Payload is actually having sub-fields for client credentials (id, secret) in yaml:

  enhancer: 'Bearer '
  payload:
    grant_type: client_credentials
    client_id: client_id
    client_secret: client_secret

Reply:
`grant_type` and `Bearer` are fixed values and I don't see any reason they could be changed. I can see why you put them in your config file, but thinking about the reality, the only things that could be changed and should be configurable are actually `client_id` and `client_secret`. When developing an app, it's also important to think of the ease of maintenance. I guess the question for you is, what are the fields that you really want your client to configure? If you feel Bearer can be changed, is your script able to hander the different auth methods?

Reply:
Got your point. Thanks!